name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/editor/ClientInput.java

```diff
@@ -529,17 +535,88 @@ private File getPreferenceStoreFile(File file) throws IOException
         }
     }
 
+    private Configuration getSelectedReportingPeriodConfiguration()
+    {
+        return client.getSettings().getConfigurationSet(WellKnownConfigurationSets.MISCELLANEOUS) //
+                        .computeIfAbsent(SELECTED_REPORTING_PERIOD_KEY,
+                                        u -> new Configuration(u, "", getReportingPeriods().get(0).getCode())); //$NON-NLS-1$
+    }
+
+    public void storeSeletedReportingPeriod(ReportingPeriod period)
+    {
+        storeSelectedReportingPeriod(period.getCode());
+    }
+
+    /* package */ void storeSelectedReportingPeriod(String code)
+    {
+        getSelectedReportingPeriodConfiguration().setData(code);
+    }
+
+    public ReportingPeriod getSeletedReportingPeriod()
+    {
+        var config = getSelectedReportingPeriodConfiguration();
+
+        try
+        {
+            String code = config.getData();
+            if (code != null)
+                return ReportingPeriod.from(code);
+        }
+        catch (IOException ignore)
+        {
+            PortfolioPlugin.log(ignore);
+        }
+
+        return getReportingPeriods().get(0); // there should be at least one
+        // element
+    }
+
     public List<ReportingPeriod> getReportingPeriods()
     {
         if (reportingPeriods != null)
             return reportingPeriods;
 
+        List<ReportingPeriod> answer = stringToReportingPeriods(getReportingPeriodsConfiguration().getData());
+
+        if (answer.isEmpty())
+        {
+            answer = defaultReportingPeriods();
+        }
+
+        reportingPeriods = answer;
+        return answer;
+    }
+
+    private Configuration getReportingPeriodsConfiguration()
+    {
+        return client.getSettings().getConfigurationSet(WellKnownConfigurationSets.MISCELLANEOUS) //
+                        .computeIfAbsent(REPORTING_PERIODS_KEY, u -> new Configuration(u, "", //$NON-NLS-1$
+                                        reportingPeriodsToConfigurationString(defaultReportingPeriods())));
+    }
+
+    private static List<ReportingPeriod> defaultReportingPeriods()
+    {
         List<ReportingPeriod> answer = new ArrayList<>();
+        for (int ii = 1; ii <= 3; ii++)
+            answer.add(new ReportingPeriod.LastX(ii, 0));
+        return answer;
+    }
+
+    private static String reportingPeriodsToConfigurationString(List<ReportingPeriod> periods)
+    {
+        StringBuilder buf = new StringBuilder();
+        for (ReportingPeriod p : periods)
+            buf.append(p.getCode()).append(';');
```

Comment:
As we use the configuration sets for storing the data, I wonder if we do not need to concatenate the reporting periods anymore. We could have a configuration set "REPORTING_PERIODS". Each configuration is one reporting period. I understand that we do not really need all the different property (uuid, name, data), but then let's leave them empty.

What do you think?

Reply:
I also thought about this, but decied to let this as it is . Because current serialization seems light a lightwight solution.
But if its okay to use separate ``Configuration``s for each period, then I can change this.